lib/routes/account.js

```diff
@@ -1004,26 +1078,35 @@ module.exports = function (
         log.begin('Account.RecoveryEmailResend', request)
         var sessionToken = request.auth.credentials
         var service = request.payload.service || request.query.service
-        if (sessionToken.emailVerified) {
+
+        // Choose which type of email to send
+        var code, func
+        if (sessionToken.emailVerified && sessionToken.tokenVerified) {
           return reply({})
+        } else if (!sessionToken.emailVerified || sessionToken.tokenVerified) {
+          code = sessionToken.emailCode
+          func = mailer.sendVerifyCode
+        } else {
+          code = sessionToken.tokenVerificationId
+          func = mailer.sendVerifyLoginEmail
```

Comment:
I believe this was my attempt to handle a possible rollback of the content server issuing a request to the new auth-server. In which case it would send the verify code email. Your approach works well too.